src/legend.jl

```diff
@@ -34,11 +34,44 @@ function add_entry!(legend::Legend, entry::String; title::String="")
 end
 
 function create_legend(scene, legend::Legend)
+    legend = remove_duplicates(legend)
     sections = legend.sections
     MakieLayout.LLegend(
         scene,
         getproperty.(sections, :plots),
         getproperty.(sections, :names),
         getproperty.(sections, :title)
     )
-end
\ No newline at end of file
+end
+
+function remove_duplicates(legend)
+    sections = legend.sections
+    titles = getproperty.(sections, :title)
+    # check if there are duplicate titles
+    unique_inds = unique_indices(titles; keep = " ")
+    has_duplicates = length(unique_inds) < length(titles)
+    # if so: remove duplicates, generate new names
+    if has_duplicates
+        sections_new = sections[unique_inds]
+    
+        names_new = map(unique_inds) do i
+            title = titles[i]
+            duplicate_inds = titles .== title
+            # ["var1", "var2"] becomes ["var1_var2"]
+            name = reduce((s,t) -> s * "_" * t, legend.names[duplicate_inds])
```

Comment:
I guess `join(legend.names[duplicate_inds], '_')` is the standard way of writing this, but I'm not completely sure we need it (`legend.names` is not used in making the graphical legend I think). I think just

```julia
names_new = legend.names[unique_inds]
```

is fine.